build-tests/heft-example-plugin-02/src/index.ts

```diff
@@ -0,0 +1,20 @@
+import { IHeftPlugin, HeftSession, HeftConfiguration } from '@rushstack/heft';
+import { default as heftExamplePlugin01 } from 'heft-example-plugin-01';
+
+const PLUGIN_NAME: string = 'example-plugin-02';
+
+export class ExamplePlugin02 implements IHeftPlugin {
+  public displayName: string = PLUGIN_NAME;
+
+  public apply(heftSession: HeftSession, heftConfiguration: HeftConfiguration): void {
+    heftSession.applyForPlugin(heftExamplePlugin01, (plugin: typeof heftExamplePlugin01) => {
```

Comment:
What happens if `heftExamplePlugin01` never actually gets loaded?  Is there any way for `ExamplePlugin02` to determine that?  Or would it simply wait indefinitely in hopes that it will eventually get loaded?

Reply:
`heftExamplePlugin01` is an object reference, so this problem gets even more interesting in the case of side-by-side versions, where the plugin does get loaded, but with a different `prototype`.

Reply:
The callback just won't get called. That's a scenario I'd expect the plugin developer(s) to have to handle themselves.

Reply:
The side-by-side version issue is something I thought about, but I don't have a good solution for.

Reply:
@iclanton What if we put some stakes in the ground like this:

- When a plugin hooks into another plugin, there should be a formal API contract that we try to keep stable.  This means:
- Plugins cannot be hooked into unless they explicitly agree to this in their manifest
- Plugins should explicitly separate their "public API" from the internal implementation details.
- Only Heft can run the `new ExamplePlugin02();` code; the consumer must request the public API object from Heft.  In a side-by-side scenario, this ensures the correct instances is returned.